File: pkg/web/es.go
package web

// ES基础操作 路由
func (this *WebServer) runEs() {

	const AbsolutePath = "/api/es"
	group := this.engine.Group("ES基础操作", AbsolutePath)
	{

		group.POST(false, "Ping连接", "/PingAction", this.esController.PingAction)
		group.POST(false, "获取ES索引数", "/IndexsCountAction", this.esController.IndexsCountAction)
		group.POST(false, "进行ES的Cat操作", "/CatAction", this.esController.CatAction)

	}
}

File: resources/vue/src/api/es.ts
import request from '@/utils/request'

const api = '/api/es/'

export function PingAction(data) {
  return request({
    url: api + `PingAction`,
    method: 'post',
    data
  })
}

export function IndexsCountAction(data) {
  return request({
    url: api + `IndexsCountAction`,
    method: 'post',
    data
  })
}

export function CatAction(data) {
  return request({
    url: api + `CatAction`,
    method: 'post',
    data
  })
}
